packages/core/src/account/schema.ts

```diff
@@ -0,0 +1,58 @@
+import { Address as zAddress } from "abitype/zod";
+import type { Chain, Transport } from "viem";
+import z from "zod";
+import type { PublicErc4337Client, SupportedTransports } from "../client/types";
+import type { SmartAccountSigner } from "../signer/types";
+import { getChain } from "../utils/index.js";
+
+export const BaseSmartAccountParamsSchema = <
+  TTransport extends SupportedTransports = Transport
+>() =>
```

Comment:
oh I see why it's `ReturnType` for this... this is a lambda. Is this the only way to do generics on a zod schema?

Reply:
cleanest way I saw how to.